sympy/functions/special/gamma_functions.py

```diff
@@ -583,6 +583,12 @@ class polygamma(Function):
     .. [4] http://functions.wolfram.com/GammaBetaErf/PolyGamma2/
     """
 
+    def _eval_evalf(self, prec):
+        n = self.args[0]
+        # the mpmath polygamma implementation valid only for nonnegative integers
+        if n.is_real and n.is_number:
```

Comment:
How about changing the order? I think that `n.is_number` will generally take less time than `n.is_real` (though I may be wrong).

Reply:
It should be correct that `n.is_number` is faster as it only has to check if there are any free symbols and not worry about the assumptions on the symbols.